bycycle/group/features.py

```diff
@@ -74,91 +91,154 @@ def compute_features_2d(sigs, fs, f_range, compute_features_kwargs=None,
+
+            # Shift sample indices
+            sample_cols = [col for col in df_single.columns if 'sample_' in col]
+
+            for col in sample_cols:
+                df_single[col] = df_single[col] - first_idx
+
+            df_features.append(df_single)
+
+        # Apply different thresholds if specified
+        if len(kwargs) > 0:
+
+            for idx, compute_kwargs in enumerate(kwargs):
+
+                burst_method = compute_kwargs.pop('burst_method', 'cycles')
+                thresholds = compute_kwargs.pop('threshold_kwargs', {})
+                center_extrema_next = compute_kwargs.pop('center_extrema', None)
 
-    if return_samples is True:
-        return df_features, df_samples
+                if idx > 0 and center_extrema_next is not None \
+                    and center_extrema_next != center_extrema:
+
+                    warnings.warn('''
+                        The same center extrema must be used when axis is None and
+                        compute_features_kwargs is a list. Proceeding using the first
+                        center_extrema: {extrema}.'''.format(extrema=center_extrema))
+
+                if burst_method == 'cycles':
+                    df_features[idx] = detect_bursts_cycles(df_features[idx], **thresholds)
+
+                elif burst_method == 'amp':
+                    df_features[idx] = detect_bursts_amp(df_features[idx], **thresholds)
+
+    else:
+        raise ValueError("The axis kwarg must be either 0 or None.")
 
     return df_features
 
 
-def compute_features_3d(sigs, fs, f_range, compute_features_kwargs=None,
+def compute_features_3d(sigs, fs, f_range, compute_features_kwargs=None, axis=0,
                         return_samples=True, n_jobs=-1, progress=None):
     """Compute shape and burst features for a 3 dimensional array of signals.
 
     Parameters
     ----------
     sigs : 3d array
-        Voltage time series, with shape [n_groups, n_signals, n_points].
+        Voltage time series, with 3d shape, i.e. (n_channels, n_epochs, n_samples).
     fs : float
         Sampling rate, in Hz.
     f_range : tuple of (float, float)
         Frequency range for narrowband signal of interest, in Hz.
     compute_features_kwargs : dict or 1d list of dict or 2d list of dict
         Keyword arguments used in :func:`~.compute_features`.
+    axis : {0, 1, None}, optional, default: 1
+        Which axes to calculate features across:
+
+
+        - ``axis = 0`` : Features are computed for each signal independently
+          across the first axis, i.e. across channels in (n_channels, n_epochs, n_timepoints).
+        - ``axis = 1`` (default) : Features are computed for each signal independently
+          across the second axis, i.e. across channels in (n_epochs, n_channels, n_timepoints).
+        - ``axis = None`` : Features are computed across a flattened 1d array, this is typically
+          not recommended unless the 3d array was recored continuously.
```

Comment:
Ohhh, wait. it's only when I finished reading the code that I realized this flattens -> 2D, and not all the way to 1D. 
in that case, this option feels equally sensible to the flatten option within `compute_features_2d`. I'm not sure either are super general, but I think they can plausibly be kept. But - this doc should be clear as to how it flattens.

Reply:
The 3D array does eventually get flatten to 1D (within `compute_features_2d`), when `axis=None`. The 2d func is used in this case to maintain a 2D dataframe structure that can more easily be reorganized into a 3D list of dataframes. I added this option to make the function as flexible as possible for all use cases. But now that you mention it, I don't think this would ever be useful and can probably be removed.